phpstan.neon.dist

```diff
@@ -0,0 +1,49 @@
+parameters:
+    level: 3
+    paths:
+        - %currentWorkingDirectory%/lib
+    autoload_files:
+        - %currentWorkingDirectory%/tests/phpstan-polyfill.php
+    reportUnmatchedIgnoredErrors: false
+    ignoreErrors:
+        # extension not available
+        - '~^(Used )?(Function|Constant) sasql_\S+ not found\.\z~i'
+
+        # removing it would be BC break
+        - '~^Constructor of class Doctrine\\DBAL\\Schema\\Table has an unused parameter \$idGeneratorType\.\z~'
+
+        # changing these would be a BC break, to be done in next major
+        - '~^Method Doctrine\\DBAL\\Driver\\IBMDB2\\DB2Connection::exec\(\) should return int but returns bool\.\z~'
+        - '~^Method Doctrine\\DBAL\\Query\\QueryBuilder::execute\(\) should return Doctrine\\DBAL\\Driver\\Statement\|int but returns Doctrine\\DBAL\\Driver\\ResultStatement\.\z~'
+        - '~^Property Doctrine\\DBAL\\Schema\\Table::\$_primaryKeyName \(string\) does not accept (default value of type )?false\.\z~'
+        - '~^Property Doctrine\\DBAL\\Schema\\Schema::\$_schemaConfig \(Doctrine\\DBAL\\Schema\\SchemaConfig\) does not accept default value of type false\.\z~'
+        - '~^Method Doctrine\\DBAL\\Schema\\ForeignKeyConstraint::onEvent\(\) should return string\|null but returns false\.\z~'
+        - '~^Method Doctrine\\DBAL\\Schema\\(Oracle|PostgreSql|SQLServer)SchemaManager::_getPortableTableDefinition\(\) should return array but returns string\.\z~'
+        - '~^Method Doctrine\\DBAL\\Platforms\\(|SQLAnywhere|Sqlite)Platform::_getTransactionIsolationLevelSQL\(\) should return string but returns int\.\z~'
+        - '~^Method Doctrine\\DBAL\\Driver\\OCI8\\OCI8Connection::lastInsertId\(\) should return string but returns (int|false)\.\z~'
+        - '~^Method Doctrine\\DBAL\\Driver\\SQLSrv\\SQLSrvConnection::errorCode\(\) should return string\|null but returns false\.\z~'
+
+        # actually OCI-Lob - not even a valid class name...
+        - '~^Call to an undefined method object::writeTemporary\(\)\.\z~'
```

Comment:
Hmm… all generated stubs have it as `OCI_Lob` (probably to make it valid), shouldn't then PHPStan just use the "fixed" names? The `oci_new_descriptor()` is stubbed as `oci_new_descriptor() : OCI_Lob`, and `OCI_Lob::writeTemporary()` is also stubbed, so I'd expect not seeing this error.

Reply:
https://github.com/phpstan/phpstan/issues/1123

Reply:
Not really sure what to think about this. :)